scripts/verify-payouts.py

```diff
@@ -0,0 +1,100 @@
+
+# ========================================= MAIN EXECUTION =========================================
+
+# Get expected total match amount
+expected_total_dai_amount = Decimal(0)
+for (index,payout) in enumerate(expected_payouts):
+  # Skip the first one since we intentionally duplicated it in our payment mapping to ensure it gets
+  # filtered out when parsing event logs later
+  if index != 0:
+    expected_total_dai_amount += Decimal(payout['amount'])
+expected_total_dai_amount = expected_total_dai_amount / SCALE
+
+# Get contract instances
+w3 = Web3(Web3.WebsocketProvider(PROVIDER))
+match_payouts = w3.eth.contract(address=match_payouts_address, abi=match_payouts_abi)
+dai = w3.eth.contract(address=dai_address, abi=erc20_abi)
+
+# Get PayoutAdded events
+payout_added_filter = match_payouts.events.PayoutAdded.createFilter(fromBlock=from_block)
+payout_added_logs = payout_added_filter.get_all_entries() # print these if you need to inspect them
+
+# Get total required DAI balance based on PayoutAdded events. Events will be sorted chronologically,
```

Comment:
Is returning events chronologically a documented web3 feature, or are we counting on observed behavior from the RPC or the library that could change out from under us? Might be worth adding a defensive sort-by-date.

Reply:
It's using `eth_getFilterLogs`, but the JSON RPC docs say nothing about the return order, so I'm guessing it's not specified. 

As of 0c16c49ff0582ca6ba85537a4a489cbd265d8401 we explicitly sort by block, and throw an error if a recipient is set twice in the same block